### PR TITLE
Added hook for ham menu functionality

### LINKED ISSUE #18: Hamburger menu does not work on mobile
By default, Bulma doesn't include any Javascript. Because of this, the navbar when collapsed to a menu doesn't actually work on mobile. Using the documentation from the [Bulma navigation toggle](https://bulma.io/documentation/components/navbar/#navbarJsExample), this functionality should be implemented.

Ideally, it should be more than just a class toggle; Try to add a sliding animation of some sort as well.

### PR DESCRIPTION
**Pull request by asharonbaltazar**
Edit: Fixes #18 
The menu is implemented with toggle functionality and works as expected. No need for custom JS — possible with a `useState` hook

### CODE REVIEW

**Review by arnavb — CHANGES_REQUESTED** [review]
Functionality-wise, this seems good. Could you also address the other part of the issue, the CSS transition for the dropdown?

**Comment by asharonbaltazar** [comment]
Sure. Any particular way you'd like me to achieve that?

**Comment by arnavb** [comment]
I think adding a CSS transition to the active class would be the best approach. This [issue](https://github.com/jgthms/bulma/issues/1790) has some examples.

**Comment by asharonbaltazar** [comment]
I think CSS is too messy, IMO, especially with a pre-existing library. React has plenty of animation libraries, though. 

I'll just stop here. You can either accept my PR or just close it. Sorry about that.

File: src/App.js [hunk 1]
```diff
@@ -21,13 +23,14 @@ export default function App() {
             aria-label="menu"
             aria-expanded="false"
             data-target="navbarBasicExample"
+            onClick={() => setOpen(!open)}
```
Comment (arnavb) [inline src/App.js:26]: The button also needs the `is-active` class after the toggle.

**Review by arnavb — CHANGES_REQUESTED** [review]
As far as CSS transition goes, it really isn't going to bloat anything since it's just simple code to add (I wrote the code below by tweaking the code from that issue a little bit). Try doing something similar and adding it. In contrast to what you said, a React transition library will add _much_ more bloat than just 2 CSS rule blocks.

```CSS
.navbar-menu {
  display: block;
  opacity: 0;

  position: absolute;
  left: 0;
  right: 0;

  transform: translateY(-100%);
  transition: all 0.2s cubic-bezier(0.3, 0.01, 1, -0.09);
  pointer-events: none;
}

.navbar-menu.is-active {
  opacity: 1;
  transform: none;
  pointer-events: auto;
}
```

**Review by arnavb — APPROVED** [review]
Even though the transition itself is incomplete, I'm merging this in because someone else will work on getting that set up.

**Pull request merged**